server/routes_test.go

```diff
@@ -53,6 +53,55 @@ func TestRouteConfig(t *testing.T) {
 	}
 }
 
+func TestClientRemoveSubsOnDisconnectFromCluster(t *testing.T) {
+	optsA, _ := ProcessConfigFile("./configs/srv_a.conf")
+	optsB, _ := ProcessConfigFile("./configs/srv_b.conf")
+
+	optsA.NoSigs, optsA.NoLog = true, true
+	optsB.NoSigs, optsB.NoLog = true, true
+
+	srvA := RunServer(optsA)
+	defer srvA.Shutdown()
+
+	srvB := RunServer(optsB)
+	defer srvB.Shutdown()
+
+	cluster := []*Server{srvA, srvB}
+
+	// Wait for route to form.
+	checkClusterFor(t, routesCheck(len(cluster)-1), 10*time.Second, cluster...)
```

Comment:
Maybe https://github.com/nats-io/gnatsd/blob/master/test/cluster_test.go is better place for this test, if we test the behavior of the cluster? I can move it in this PR